src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -104,39 +105,39 @@
 using namespace cldnn;
 using namespace ov::intel_gpu;
 
-static void adjust_num_cores(ov::threading::IStreamsExecutor::Config& config) {
-    if (ov::get_available_cores_types().size() == 1) {
-        return;
-    }
-
-    const auto total_num_cores = ov::get_number_of_logical_cpu_cores();
-    const auto total_num_big_cores = ov::get_number_of_logical_cpu_cores(true);
-    const auto total_num_little_cores = total_num_cores - total_num_big_cores;
-    auto core_type = config._threadPreferredCoreType;
-
-    int num_cores = total_num_cores;
-    if (core_type == ov::threading::IStreamsExecutor::Config::BIG) {
-        num_cores = total_num_big_cores;
-    } else if (core_type == ov::threading::IStreamsExecutor::Config::LITTLE) {
-        num_cores = total_num_little_cores;
-    }
-
-    config._streams = std::min(config._streams, num_cores);
-}
-
 static ov::threading::IStreamsExecutor::Config make_task_executor_config(const ExecutionConfig& config, std::string tags, int num_streams = 0) {
     ov::threading::IStreamsExecutor::Config task_executor_config(tags, 1);
     task_executor_config._streams = (num_streams > 0) ? num_streams : config.get_property(ov::compilation_num_threads);
     auto priority = config.get_property(ov::intel_gpu::hint::host_task_priority);
     switch (priority) {
-        case ov::hint::Priority::LOW: task_executor_config._threadPreferredCoreType = ov::threading::IStreamsExecutor::Config::LITTLE; break;
-        case ov::hint::Priority::MEDIUM: task_executor_config._threadPreferredCoreType = ov::threading::IStreamsExecutor::Config::ANY; break;
-        case ov::hint::Priority::HIGH: task_executor_config._threadPreferredCoreType = ov::threading::IStreamsExecutor::Config::BIG; break;
-        default: OPENVINO_ASSERT(false, "[GPU] Can't create task executor: invalid host task priority value: ", priority);
+    case ov::hint::Priority::LOW:
+        task_executor_config = ov::threading::IStreamsExecutor::Config::update_executor_config(
+            task_executor_config,
+            task_executor_config._streams,
+            1,
+            ov::threading::IStreamsExecutor::Config::LITTLE,
+            false);
+        break;
+    case ov::hint::Priority::MEDIUM:
+        task_executor_config = ov::threading::IStreamsExecutor::Config::update_executor_config(
+            task_executor_config,
+            task_executor_config._streams,
+            1,
+            ov::threading::IStreamsExecutor::Config::ANY,
+            false);
+        break;
+    case ov::hint::Priority::HIGH:
+        task_executor_config = ov::threading::IStreamsExecutor::Config::update_executor_config(
```

Comment:
I think this part can be refactored to avoid code duplication:
```cpp
auto preferred_core_type = ov::threading::IStreamsExecutor::Config::ANY;
switch (priority) {
        case ov::hint::Priority::LOW: preferred_core_type = ov::threading::IStreamsExecutor::Config::LITTLE; break;
        case ov::hint::Priority::MEDIUM: preferred_core_type = ov::threading::IStreamsExecutor::Config::ANY; break;
        case ov::hint::Priority::HIGH: preferred_core_type = ov::threading::IStreamsExecutor::Config::BIG; break;
        default: OPENVINO_ASSERT(false, "[GPU] Can't create task executor: invalid host task priority value: ", priority);
}
task_executor_config = ov::threading::IStreamsExecutor::Config::update_executor_config(
    task_executor_config,
    task_executor_config._streams,
    1,
    preferred_core_type,
    false);
```

Reply:
thanks, updated